apps/app/src/modules/game/dice/components/drag-drop-dices.tsx

```diff
@@ -0,0 +1,73 @@
+import { DiceZoneEnum } from '@yams-tactics/domain';
+import { DragDropContext, OnDragEndResponder } from '@hello-pangea/dnd';
+
+export function DragDropDices({
+  setLockedDices,
+  lockedDices,
+  setDicesOrder,
+  dicesOrder,
+  setIsDragging,
+  children,
+}: {
+  setLockedDices: (lockedDices: string[]) => void;
+  lockedDices: string[];
+  setDicesOrder: (dicesOrder: string[]) => void;
+  dicesOrder: string[];
+  setIsDragging: (isDragging: boolean) => void;
+  children: JSX.Element;
+}) {
+  const onDragEnd: OnDragEndResponder = (result) => {
+    const { destination, source } = result;
+    setIsDragging(false);
+
+    if (!destination) {
+      return;
+    }
+
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return;
+    }
+
+    // TODO refactor this shit
+    if (destination.droppableId === source.droppableId) {
+      if (destination.droppableId === DiceZoneEnum.playing) {
+        const element = dicesOrder[source.index];
+        dicesOrder.splice(source.index, 1);
+        dicesOrder.splice(destination.index, 0, element);
+        setDicesOrder(dicesOrder);
+      } else {
+        const element = lockedDices[source.index];
+        lockedDices.splice(source.index, 1);
+        lockedDices.splice(destination.index, 0, element);
+        setLockedDices(lockedDices);
+      }
+    } else {
+      if (destination.droppableId === DiceZoneEnum.playing) {
+        const element = lockedDices[source.index];
+        lockedDices.splice(source.index, 1);
+        setLockedDices(lockedDices);
+        dicesOrder.splice(destination.index, 0, element);
+        setDicesOrder(dicesOrder);
+      } else {
```

Comment:
This could be it's own hook :thumbs:
This will allow to limit props drilling into this component from dice zone if we also return dice zone state from this hook.

```typescript
useDices(player: PlayerModel): { dices: (DiceModel & {locked: boolean})[] /* (ordered) */ }
```